src/ResourceManager/Network/Commands.Network/VirtualNetworkGatewayConnection/NewAzureRmIpsecPolicyCommand.cs

```diff
@@ -125,8 +125,10 @@ public override void Execute()
             base.Execute();
             var ipsecPolicy = new PSIpsecPolicy();
 
-            ipsecPolicy.SALifeTimeSeconds = this.SALifeTimeSeconds;
-            ipsecPolicy.SADataSizeKilobytes = this.SADataSizeKilobytes;
+            // default SA values
+            ipsecPolicy.SALifeTimeSeconds = (this.SALifeTimeSeconds == 0) ? 27000 : this.SALifeTimeSeconds;
+            ipsecPolicy.SADataSizeKilobytes = (this.SADataSizeKilobytes == 0) ? 102400000 : this.SADataSizeKilobytes;
```

Comment:
@henry416 rather than using `this.X == 0` as the check for if the parameter was provided, can we use `!this.MyInvocation.BoundParameters.ContainsKey("X")` ?